android/src/main/java/com/gr4vy/embedreactnative/EmbedReactNativeModule.java

```diff
@@ -120,6 +143,36 @@ private static WritableArray convertCartItemsToJson(ReadableArray cartItemsArray
       cartItemWritableMap.putString("name", cartItemMap.getString("name"));
       cartItemWritableMap.putInt("quantity", cartItemMap.getInt("quantity"));
       cartItemWritableMap.putInt("unitAmount", cartItemMap.getInt("unitAmount"));
+
+      List<String> optionalProps = new ArrayList<String>(Arrays.asList(
+        "discountAmount",
+        "taxAmount",
+        "externalIdentifier",
+        "sku",
+        "productUrl",
+        "imageUrl",
+        "categories",
+        "productType"
+      ));
+      for (String prop : optionalProps) {
+        Object value = getMapOptionalValue(cartItemMap, prop);
+        if (value != null) {
+          switch (value.getClass().getSimpleName()) {
+            case "Integer":
+              cartItemWritableMap.putInt(prop, (Integer) value);
+              break;
```

Comment:
As we're already doing the type check in the private function, maybe we could also populate the other map in there:
```java
  public static void setOptionalValue(ReadableMap source, WritableArray target, String key) {
    if (map.hasKey(key)) {
      if (!map.isNull(key)) {
        switch (map.getType(key)) {
          case String:
            return target.putString(key, source.getString(key));
          etc.
        }
      }
    }
  }
```

So here you just need to do:
```java
for (String prop : optionalProps) {
    setOptionalValue(cartItemMap, cartItemWritableMap, prop);
}
```

Reply:
great, updated thanks